src/azmi-main/Operations.cs

```diff
@@ -205,5 +210,46 @@ public string setBlob_byBlob(string filePath, string blobUri, string identity =
                 throw IdentityError(identity, ex);
             }
         }
+
+        // Validate URL to a key vault secret is generally valid
+        private void validateSecretURL(string secretIdentifierUrl, out Uri keyVaultUri, out Uri secretIdentifierUri)
+        {
+            // Example of expected URL: https://mine-key-vault.vault.azure.net/secrets/mineSecret.pwd
+            if (!Uri.IsWellFormedUriString(secretIdentifierUrl, UriKind.Absolute))
+                throw new UriFormatException($"Provided URL '{secretIdentifierUrl}' is not well formed URL.");
```

Comment:
I am not sure if we need this. The same exception will be raised with Uri constructor in the next line. Up to you.